### ISSUE TITLE
Call with Grant 4-5-18 IMPORTANT

### ISSUE THREAD

**Issue by kdjstudios** [kdjstudios/SABillingGitlab#797]
In GitLab by @kdjstudios on Apr 5, 2018, 12:04

HD Ticket: http://www.servicedesk.answernet.com/profiles/ticket/2018-04-06-62119/conversation

Hello team,

I just had a phone call with Grant to review all of his concerns, and keep pushing him towards running his billing cycle. He did have a few concerns that he would like addressed before he runs the cycle. I will list those below. 

- Update all parent accounts to have the "Separate Child Account Transaction" selected. - This I would presume is a quick update we can do to the database. 
- Update the Rounding of charges. As @tim.traylor mentioned during the call, we want to use the ceil/floor/round correctly. Posting update in #767 

May we please do the two items right away! I hope to have this ready for 4/6/18 So we may have him do his billing cycle before the weekend. As it was supposed to be done on April 4th.

**Comment by kdjstudios** [comment]
In GitLab by @kdjstudios on Apr 5, 2018, 12:05

changed title from **Call with Grant 4-5-18** to **Call with Grant 4-5-18{+ IMPORTANT+}**

**Comment by kdjstudios** [comment]
In GitLab by @smasih on Apr 6, 2018, 08:50

@kdjstudios 

> Update all parent accounts to have the "Separate Child Account Transaction" selected. - This I would presume is a quick update we can do to the database.

Updated all the parent accounts



> Update the Rounding of charges. As @tim.traylor mentioned during the call, we want to use the ceil/floor/round correctly. Posting update in #767

Updated accordingly (removed the penny round up line)

**Comment by kdjstudios** [comment]
In GitLab by @kdjstudios on Apr 6, 2018, 09:18

@smasih 

Thank you, I will have Grant begin and hopefully finish processing his billing today.

**Comment by kdjstudios** [comment]
In GitLab by @kdjstudios on Apr 6, 2018, 11:20

I called Grant and we tried to walk through creating a new Billing Cycle, however it does not work.

When we click the "create" button it says a cycle is created, but does not go to the next step. 

Grant was extremely unimpressed with the amount of issues they are having using SA Billing. I informed him that this was probably due to the last minute changes he requested yesterday and that the support team will have this resolved soon so we can finish processing the billing cycle.

**Comment by kdjstudios** [comment]
In GitLab by @kdjstudios on Apr 6, 2018, 12:18

@smasih @tim.traylor @thomas.gentry @pchaudhary @rchauhan 

**THIS IS HORRIFIC!!!!!!**

In WhatsApp chat you stated this was fixed. **IT IS NOT**. And appears to be way worse then it was before. 

I called Grant back and we were presented with the step to upload the usage file. I asked him to wait a second, while I checked the last cycles invoices. All reports did not work, and there was no data in them!
- What happened to their first billing cycle information? This is critical to running a system in parallel!!!

I posted in WhatsApp the issue, with no response for over 10 minutes. So Grant and I proceeded with running the billing cycle. This time we were able to upload and process the cycle up to the Draft invoices step where, we stopped to run some initial reports and do some comparisons. However, all the reports are now for the wrong billing cycle and there is no billing cycle for their march invoices. 


- WHAT HAPPENED?
- THIS NEEDS TO BE FIXED ASAP

**Comment by kdjstudios** [comment]
In GitLab by @kdjstudios on Apr 6, 2018, 12:22

The only positive thing about this, is luckily they did not go live...

Grant is extremely unimpressed.  He stated they are having issue after issue, reports not working, data being missing, and how slow SA Billing is. I profusely apologized to Grant for the issues, and informed him that I would be having the support team fix everything before we continue to process the rest of his billing.

**Comment by kdjstudios** [comment]
In GitLab by @kdjstudios on Apr 6, 2018, 13:02

Update from @smasih 

> We are restoring yesterday's db to get the old cycle back and then we will upload the new one on top of it

Email to Grant:

> The support team is currently looking into the issues we discovered while processing the billing cycle this morning. Their initial plan to resolve this, is to restore the database to last night and then they will reprocess the billing cycle, and be able to follow any errors in real time.
>
> In order to do this they will need the usage file. Would you please reply with this billing cycle’s usage file? Thanks.

Response from Grant:

> Usage file for April 4, 2018
>
> [TUSACC04.fla](/uploads/fbeb1b416b63bde1cf2d2f5a0da26f48/TUSACC04.fla)

**Comment by kdjstudios** [comment]
In GitLab by @kdjstudios on Apr 6, 2018, 14:51

update from @smasih 

> It's bad, system stopped taking backup after 20 Feb
> 
> So unfortunately we don't have a backup 
> 
> Besides there are no invoice in the system beside the new ones grant creted today (which are in draft stage)
> 
> Did we revert the billing cycle..? Because if we did the old invoices  are lost ..
> And only way to fix that is to first upload the last usage file ... Finalize it and then upload the new one.

My response:

> All me and Grant did was try to create the next billing cycle, then you guys did somethign to get it to the point to upload the file, then we continued processing.

@smasih wrote:

> Ok, thanks for the info
> Now we have a situation ... Because we don't have backup  after 6 March so we need the old usage file as well to recreate the older one and then the new on on top of it.
> 
> Luckily we have values previous balance line item  so we can at least check the discrepancy for this cycle.

I wrote:

> That is not good at all, we would be missing payments, staged fees, credits, etc basicly everything that SCC has been doing for the last two months right?

@smasih responded:

> Yes we will be missing on all of it ... It's not good ....
> 
> I don't know why but system stopped taking the db backup after 20 March ... Or we could have restored all the details.

**Comment by kdjstudios** [comment]
In GitLab by @kdjstudios on Apr 6, 2018, 14:52

@smasih 

Can you please clarify was it March 20th, March 6th, or February 20th was the last backup?

**Comment by kdjstudios** [comment]
In GitLab by @smasih on Apr 6, 2018, 15:02

@kdjstudios 

> Can you please clarify was it March 20th, March 6th, or February 20th was the last backup?

We have backup dated March 6^th (this one we took manually on march 6)

Usually system takes a backup by default, but it stopped taking that backup after feb 20^th on dev server (we have the backup folders but those are empty after feb 20^th )

**Comment by kdjstudios** [comment]
In GitLab by @kdjstudios on Apr 6, 2018, 15:23

changed the description

**Comment by kdjstudios** [comment]
In GitLab by @kdjstudios on Apr 6, 2018, 15:25

Update sent to Grant:

> The support team has not been able to find a resolution to this and is still researching it. it appears that the issue is much larger than initially anticipated. I currently do not have any ETA on when this will be resolved; but as soon as I know more I will update you.

**Comment by kdjstudios** [comment]
In GitLab by @kdjstudios on Apr 9, 2018, 11:45

@smasih @pchaudhary @tim.traylor @rchauhan 

Just received a call from Garrett. Where are we on this? 

- What happened?
- What are we doing to fix it?
- What do we need from SCC, if anything?
- When will this be ready for them?

Garrett is extremely unhappy with the issues they continue to have and now having to wait on being able to use SA Billing is pushing back everything and causing more issues.

**Comment by kdjstudios** [comment]
In GitLab by @tim.traylor on Apr 9, 2018, 11:53

@smasih @kdjstudios Hi Kyle & Sumeet, was there going to be any way to catch up from the 3/6 backup, or do we just have to start over with a fresh data load?

**Comment by kdjstudios** [comment]
In GitLab by @tim.traylor on Apr 9, 2018, 12:31

@kdjstudios @smasih OK.  IN talking with Kyle it sounds like they will be too far behind to make this work.  We're going to need to get a fresh import and get it loaded ASAP, and then they can parallel this month.

@kdjstudios Kyle, please let them know that we apologize, but that their parallel testing was running on a non-production server and the backup failure was not being monitored. We will correct this going forward, but understand the inconvenience it has caused.  Thx, Tim

**Comment by kdjstudios** [comment]
In GitLab by @tim.traylor on Apr 9, 2018, 12:32

@smasih @pchaudhary Hi Guys, Also on this, do we not have a mechanism that can inform us that the backups are failing?  

Since we have to have you run another import I asked Kyle to post the list of issues that we've had, so they can hopefully be avoided this last time.
 
Thx, Tim

**Comment by kdjstudios** [comment]
In GitLab by @kdjstudios on Apr 9, 2018, 12:54

Email sent to Grant and Garrett:

> After reviewing with our support team, they found out that the backups on our non-production server were not running as often as anticipated. Since this was not on a production server they were also not being monitored as closely. So after Grant and I incurred the issues on Friday, April 6th; when they went to restore from a backup and could not find one. The last backup they found was dated March 6th. We could restore from that data back up; but in order to save your team time and work for having to enter data from that far back. We would like to start from a fresh import to also insure all data matches exactly what is in your system. 
> 
> In order to start your parallel testing from a fresh import; we will need all of your export files from your system. Please send over all of the files and our support team will be able to import those within 24 hours of receiving them. Once they have been imported you can continue to run parallel and your upcoming billing cycle can verify for any variances. 
> 
> We apologize for any inconvenience this has caused you and your team. Please respond to this email if you have any concerns or questions.

**Comment by kdjstudios** [comment]
In GitLab by @kdjstudios on Apr 9, 2018, 13:13

**Main Issues found during previous imports:**

- Parent Child accounts with same Resource ID #728 and #707 
- Separate out child transactions #783 
- Child accounts missing their charges #739 and #738 
- Tax rates and associations incorrect #778 , #747 , and #785  
- Print and Mail needs to be pull bills #707 
- Opening balance incorrect #749 ; #747 ; and #751 
- Possible missing additional fees #714 


**Also be sure to check other concerns such as:**

- New User logins #748 
- Reports not working #745 
- Login issues #720 
- Drop down search #723

**Comment by kdjstudios** [comment]
In GitLab by @pchaudhary on Apr 10, 2018, 01:23

@kdjstudios 

Thanks kyle for consolidating all the SCC tickets, appreciate the help.

**Comment by kdjstudios** [comment]
In GitLab by @kdjstudios on Apr 10, 2018, 08:29

@pchaudhary @smasih @tim.traylor 

Update sent to Grant to clarify which files are needed. Please let me know if I missed any.

We are needing all of your export files. I will list the ones that were previously sent on the last import. 

-	Billings Frequency
-	Aged detail total
-	B2rcr1
-	B2tas1
-	B2tas2
-	Invoice detail
-	B2cmr1
-	Sales tax

**Comment by kdjstudios** [comment]
In GitLab by @kdjstudios on Apr 10, 2018, 10:40

Created new ticket for the import files and its' progress: #813

**Comment by kdjstudios** [comment]
In GitLab by @smasih on Apr 23, 2018, 07:43

@kdjstudios 

can we close this ticket..?

**Comment by kdjstudios** [comment]
In GitLab by @kdjstudios on Apr 23, 2018, 10:09

@smasih 

YES! woohoo This should be the last import for SCC!!!

**Comment by kdjstudios** [comment]
In GitLab by @kdjstudios on May 30, 2018, 08:22

closed
